administrator/components/com_associations/src/Helper/AssociationsHelper.php

```diff
@@ -333,6 +334,15 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
 			$url     = $allow && $addLink ? $url : '';
 			$text    = $language->lang_code;
 
+			if (PluginHelper::isEnabled('system', 'languagecode'))
+			{
+				$pluginEnabled = PluginHelper::getPlugin('system', 'languagecode');
+				$params        = new Registry($pluginEnabled->params);
+				$code          = strtolower($language->lang_code);
+				$new_code      = $params->get($code);
+				$text          = $new_code ? $new_code : $language->lang_code;
```

Comment:
```suggestion
				$text          = $new_code ?: $language->lang_code;
```
Same for the other places there a ternary operator is used with expression 1 and 2 being the same.

Reply:
The whole if statement needs to move before the foreach and instead we only do a simple check for the text `$text = $new_code ? $new_code : $language->lang_code;`
As is it's extremely inefficient, the code keeps evaluating the same thing again and again...

Reply:
@dgrammatiko I think you are wrong because in each iteration of the loop it is a different language and language code, so it can’t be moved outside the for each loop. And even if I am wrong amenity can be moved, it also should use the Elvis operator like I suggested here because the 1st and the 2nd expression in the ternary are equal.

Reply:
Am I? 

```php
if (PluginHelper::isEnabled('system', 'languagecode'))
{
 $pluginEnabled = PluginHelper::getPlugin('system', 'languagecode');
 $params        = new Registry($pluginEnabled->params);
}

Reply:
Well if you want to move only this, then I do not see how this is related to my change suggestion here.

Reply:
It's not, sorry for placing this in the wrong line